src/main/java/chess/domain/board/RunningBoard.java

```diff
@@ -0,0 +1,128 @@
+    private static final int BLACK_PAWN_ROW = 7;
+    private static final int WHITE_PAWN_ROW = 2;
+    private static final int BLANK_START_INDEX = 3;
+    private static final int BLANK_END_INDEX = 6;
+
+    private final Map<Position, Piece> pieces;
+
+    private RunningBoard(Map<Position, Piece> pieces) {
+        this.pieces = pieces;
+    }
+
+    public static RunningBoard initiaize() {
+        Map<Position, Piece> pieces = new HashMap<>();
+        initializeBlackTeam(pieces);
+        initializeBlanks(pieces);
+        initializeWhiteTeam(pieces);
+        return new RunningBoard(pieces);
+    }
+
+
+    @Override
+    public Board movePiece(MovingFlow movingFlow) {
+        return movePiece(movingFlow, this);
+    }
+
+    @Override
+    public Piece getPiece(Position position) {
+        return pieces.get(position);
+    }
+
+    @Override
+    public Map<Position, Piece> getPieces() {
+        return pieces;
+    }
+
+    @Override
+    public boolean isNotFinished() {
+        return true;
+    }
+
+    @Override
+    public Result concludeResult() {
+        throw new IllegalStateException("게임이 끝나지 않아, 승패를 결정할 수 없습니다.");
+    }
+
+    private static void initializeBlackTeam(Map<Position, Piece> pieces) {
+        initializeEdge(pieces, Team.BLACK, BoardConfig.LINE_END);
+        initializePawns(BLACK_PAWN_ROW, Team.BLACK, pieces);
+
+    }
+
+    private static void initializeWhiteTeam(Map<Position, Piece> pieces) {
+        initializeEdge(pieces, Team.WHITE, BoardConfig.LINE_START);
+        initializePawns(WHITE_PAWN_ROW, Team.WHITE, pieces);
+    }
+
+    private static void initializeEdge(Map<Position, Piece> pieces, Team team, int edgeRow) {
+        for (int x = BoardConfig.LINE_START; x <= BoardConfig.LINE_END; x++) {
+            Position position = Position.of(x, edgeRow);
+            Piece piece = PieceFactory.createInitializedPieceWithInitialColumn(x, position, team);
+            pieces.put(position, piece);
+        }
+    }
+
+    private static void initializePawns(int row, Team team, Map<Position, Piece> pieces) {
+        for (int x = BoardConfig.LINE_START; x <= BoardConfig.LINE_END; x++) {
+            Position position = Position.of(x, row);
+            Piece initializedPawn = PieceFactory.createInitializedPiece(PieceType.INITIALIZED_PAWN, position, team);
+            pieces.put(position, initializedPawn);
+        }
+    }
+
+    private static void initializeBlanks(Map<Position, Piece> pieces) {
+        for (int y = BLANK_START_INDEX; y <= BLANK_END_INDEX; y++) {
+            initializeBlanks(pieces, y);
+        }
+    }
+
+    private static void initializeBlanks(Map<Position, Piece> pieces, int y) {
+        for (int x = BoardConfig.LINE_START; x <= BoardConfig.LINE_END; x++) {
+            Position position = Position.of(x, y);
+            Blank blank = Blank.of();
+            pieces.put(position, blank);
+        }
+    }
+
+    private Board movePiece(MovingFlow movingFlow, Board board) {
```

Comment:
```suggestion
    private Board movePiece(MovingFlow movingFlow, Board board) {
        Position from = movingFlow.getFrom();
        Position to = movingFlow.getTo();

        Piece fromPiece = board.getPiece(from);
        if (fromPiece.canNotMove(to, board)) {
            throw new IllegalArgumentException(String.format("%s 위치의 말을 %s 위치로 옮길 수 없습니다.", from, to));
        }
        Piece toPiece = board.getPiece(to);

        fromPiece = fromPiece.move(to, toPiece);
        Map<Position, Piece> pieces = updatePieces(from, to, fromPiece);
        return updateBoard(pieces, fromPiece);
    }
```
이런식으로 move가 가진 2가지 기능을 분리해보면 어떨까요
아니면 Piece가 이동가능여부를 직접 체크하는게 아니라 다른 객체로 위임해도 괜찮을 것 같네요

Reply:
한 번 PieceService를 두고, PieceService 내 move()안에 canNotMove()와 move()를 둬 봤습니다!
궁금한 게, 보통 Service로 돼있는 클래스들은 인스턴스 함수를 갖는 걸 봐왔었는데요.
제 코드에서는 딱히 상태를 가지지 않고, 변화의 가능성도 예측되지 않아 클래스로 두고 static 메서드를 갖게 했습니다.
혹시 문제가 될까요..?

Reply:
1. Service 객체의 역할은 무엇일까요? PieceService를 만들 필요가 없어보이는데 PieceService를 추가한 이유가 무엇일까요?
2. Piece가 Position 객체를 가지고 있어서 생긴 문제로 보네요. Piece가 너무 많은 책임을 가지고 있어요.
Board를 구현한 객체들도 Position을 가지고 있는데 piece가 위치 정보를 갖고 있어야 할까요?
해당 위치로 옮길 수 있는지 판단하고 board에서 처리하면 해결 할 수 있겠네요

Reply:
**[Service 객체의 역할 관련]**
Service객체는 로직을 두는 곳 정도로만 이해하고 있었습니다..!
사실, 이렇게 이해하고 있다보니 우테코에 들어오기 전 프리코스에서 불필요한 Service 객체를 만들고 모든 로직을 Service에 뒀었습니다. 우테코에서 여러 객체 간 협력에 집중해야 한다는 것을 배웠습니다.
여러가지를 배웠음에도 불구하고 이번에 Service를 둔 이유는 1. move 로직을 캡슐화 하기 위해, 2.트랜잭션의 경계를 명확히 할 수 있을 것 같아서 였습니다. move를 위해 canNotMove를 먼저 호출하고, 그다음에 move를 한다는 절차적인 로직을 숨기고 싶었습니다. 하지만 다시 생각해보니 이렇게 하더라도 캡슐화 되지 않았다는 생각이 듭니다. canNotMove, move 모두 public으로 인터페이스에 노출돼있기 때문입니다. 트랜잭션의 경우 잘 모르면서 뭔가 부가적으로 더 좋을 것 같은 것이다보니, 제가 잘못 생각하고 있는 것 같습니다. 어차피 여기서 트랜잭션 경계 설정을 하지 않고 있기도 하고요. 

구구의 피드백을 받고 생각해보니, 다른 의존성(Repository 등)과 도메인 객체 간 협력이 필요할 때 Service를 두는 것 같습니다. 현재는 Repository 등의 객체와 협력하지 않고 있으니 필요 없을 것 같다는 생각입니다.

**[Piece 내 Position 관련]**
이 부분이 굉장히 꺼림칙하던 부분이었는데 역시 제가 잘못생각한 것이었네요! 구구 말씀대로 Board도 Position을 가지고 있고, Piece도 포지션을 가지고 있는 것이 찝찝했습니다. 당시 제가 Piece에도 Position을 뒀던 이유는 객체를 보다 능동적이게 만들고 싶어서였습니다. move라는 책임이 반드시 필요하다고 생각했고, 이것을 수행할 객체를 Piece로 선정 후, 로직 수행 시 그에 맞는 데이터를 자기가 가져야 한다는 문구를 오브젝트 책에서 읽어서 Piece에 두는 것이 능동적이라고 생각했습니다.
하지만 구구의 피드백을 듣고 보니, Position을 관리하는 것은 Board이고, MoveState을 return해주는 것으로 move했다고 볼 수 있을 것 같아 Position을 Board에만 두는 것이 더 적절하다고 생각하게 됐습니다.
수정하겠습니다!